projects/module-3/rancid-tomatillos-v2.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Rancid Tomatillos
+module: 3
+tags: react, redux, testing, javascript, api
+---
+
+Have you ever watched a movie and said to yourself, "Ugh, I wish I could throw a rancid tomatillo at the screen!" Or maybe you loved the movie and you could really go for a nice, fresh tomatillo salsa to enjoy the moment. Well it's your lucky day. In this project, you'll build a movie rating site where a user can login and rate the movies they loved or hated.
+
+## Learning Goals
+
+* Reinforce React fundamentals
+* Reinforce using React Router to create a multi-page user experience
+* Reinforce component and asynchronous JS testing
+* Work with and navigate a shared, persistent API using GET, POST, and DELETE requests
+* Implement Redux as the app's place to store shared state
+* Test Redux functions
+
+## Iterations
+
+### Iteration 0 - Day-1 Deliverables
+
+Create a group Slack channel and include your instructors in the channel:
+  * Layouts/Wireframes sketches of your user interface
+  * Group DTR
+  * Project management tool (GitHub Projects, Trello, etc.) - be sure this is public so your instructors can view it
+  * PR template your group agrees to follow for every PR submitted ([here are some instructions from GitHub on how to create this](https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository))
+
+After you send us this information, we'll give you your user's name, email, and password.
+
+### Iteration 1 - User Info, API Intro, and Movies
+
+You will be assigned a user for your pair. Each user has a name, email, and password that you will use throughout the project. This is your team's user!
+
+At this point, get your head around the API's endpoints via the documentation below. Try them out in Postman - you don't need an API key for this! See what you can get from all the endpoints available.
+
+For your app, create a homepage where the visitor can see all of the movies in the database with their average ratings. When on this page, the url's path will be `/` (ex: in development, the url will look something like: `http://localhost:3000/`).
+
+### Iteration 2 - User Login
+
+Create a separate login page where your user can login. When a user navigates to this page, the URL's path should be `/login` (ex: in development, the url will be `http://localhost:3000/login`).
```

Comment:
Yeah, I think without Router this will probably lead them down unnecessary rabbit holes. I don't want them to get into the habit of updating the URL by manipulating weird stuff like browser history or whatever.

Hm. This is looking like we'll maybe need to bring in Router earlier???

Reply:
I feel like an alternative could be to have them not worry about doing any routing and just make it a single page app to start. Then [part 2](https://github.com/turingschool/front-end-curriculum/pull/443/files#diff-2cee85973b022f63ab0a9031cd3d3110R86) could involve refactoring everything to add Router in and create urls? That could be too much to pile on at the end, though.

Reply:
But it may not require us to redo much of the calendar.